lldb/tools/lldb-dap/Request/Request.h

```diff
@@ -0,0 +1,45 @@
+//===-- Request.h ---------------------------------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef LLDB_TOOLS_LLDB_DAP_REQUEST_REQUEST_H
+#define LLDB_TOOLS_LLDB_DAP_REQUEST_REQUEST_H
+
+#include "llvm/ADT/StringRef.h"
+#include "llvm/Support/JSON.h"
+
+namespace lldb_dap {
+struct DAP;
+
+class Request {
+public:
+  Request(DAP &dap) : dap(dap) {}
+  virtual ~Request() = default;
+
+  virtual void operator()(const llvm::json::Object &request) = 0;
+
+  static llvm::StringLiteral getName() { return "invalid"; };
+
+  enum LaunchMethod { Launch, Attach, AttachForSuspendedLaunch };
+
+  void SendProcessEvent(LaunchMethod launch_method);
+  void SetSourceMapFromArguments(const llvm::json::Object &arguments);
```

Comment:
I know this isn't part of your change specifically, but I do think it would be good for us to create richer types to represent some arguments like this `arguments` parameter.

That was the motivation for the creating the POD types in [Protocol.h](https://github.com/ashgti/llvm-project/blob/lldb-dap-async-refactor/lldb/tools/lldb-dap/Protocol.h) to have a clear boundary between protocol types, lldb debugger types (generally anything like `lldb::SB*`) and the helpers that are part of lldb-dap used for coordinating between the two.